aptos-move/aptos-transactional-test-harness/src/aptos_test_harness.rs

```diff
@@ -0,0 +1,916 @@
+            params.expiration_timestamp_secs,
+            ChainId::test(),
+        )
+        .sign(&private_key, Ed25519PublicKey::from(&private_key))?
+        .into_inner();
+
+        self.run_transaction(Transaction::UserTransaction(txn))?;
+
+        Ok((None, module))
+    }
+
+    fn execute_script(
+        &mut self,
+        script: CompiledScript,
+        type_args: Vec<TypeTag>,
+        signers: Vec<ParsedAddress>,
+        txn_args: Vec<MoveValue>,
+        gas_budget: Option<u64>,
+        extra_args: Self::ExtraRunArgs,
+    ) -> Result<(Option<String>, SerializedReturnValues)> {
+        let signer0 = self.compiled_state().resolve_address(&signers[0]);
+
+        if gas_budget.is_some() {
+            panic!("Cannot set gas budget for script.")
+        }
+        if extra_args.gas_unit_price.is_some() {
+            panic!("Cannot set gas price for script.")
+        }
+        if extra_args.expiration_time.is_some() {
+            panic!("Cannot set expiration time for script.")
+        }
+        if extra_args.secondary_signers.is_some() {
+            panic!("Cannot set secondary signers for script.")
+        }
+
+        let private_key = match (extra_args.private_key, &signers[0]) {
+            (Some(private_key), _) => self.resolve_private_key(&private_key),
+            (None, ParsedAddress::Named(named_addr)) => {
+                match self.private_key_mapping.get(named_addr) {
+                    Some(private_key) => private_key.clone(),
+                    None => panic_missing_private_key_named("run", named_addr.as_str()),
+                }
+            }
+            (None, ParsedAddress::Numerical(_)) => panic_missing_private_key("run"),
+        };
+
+        let mut script_blob = vec![];
+        script.serialize(&mut script_blob)?;
+
+        let params = self.fetch_transaction_parameters(
+            &signer0,
+            extra_args.sequence_number,
+            None,
+            None,
+            None,
+        )?;
+
+        let txn = RawTransaction::new_script(
+            signer0,
+            params.sequence_number,
+            TransactionScript::new(
+                script_blob,
+                type_args,
+                txn_args
+                    .into_iter()
+                    .map(|arg| TransactionArgument::try_from(arg).unwrap())
+                    .collect(),
+            ),
+            params.max_gas_amount,
+            params.gas_unit_price,
+            params.expiration_timestamp_secs,
+            ChainId::test(),
+        )
+        .sign(&private_key, Ed25519PublicKey::from(&private_key))
+        .unwrap()
+        .into_inner();
+
+        let output = self.run_transaction(Transaction::UserTransaction(txn))?;
+
+        let output = if extra_args.show_events {
+            render_events(output.events())
+        } else {
+            None
+        };
+
+        //TODO: replace this dummy value with actual txn return value
+        let a = SerializedReturnValues {
```

Comment:
I previously had this issue https://github.com/aptos-labs/aptos-core/issues/535. 
It would be ideal if AptosTestAdapter is just a wrapper around Aptos VM adapter without depending on the  MoveTestAdapter.  It would be easier when the move upgrade or our Aptos VM adpater upgrades
Not sure how complex this will be.

Reply:
I will look into the issue you reported. Getting rid of MoveTestAdapter should not be very hard, but we might need to maintain the copied/cloned code separately. Don't know if we still want to keep track of the upstream changes in case upstream has some useful updates. Sui is also using MoveTestAdapter https://github.com/MystenLabs/sui/blob/8628837f3a3dd4fc6cb5ca5dea4a90b13e926e74/crates/sui-transactional-test-runner/src/test_adapter.rs#L25. Maybe we can live with this for now. I need to spend time on improving the ergonomics of this test harness first.